subprojects/docs/src/docs/userguide/directory_layout.adoc

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[directory_layout]]
+= Directory Layout
+
+[.lead]
+Gradle uses two main directories to perform and manage its work: the <<#directory_layout:gradle_user_home_directory>> and the <<#directory_layout:project_root_directory>>.
+
+
+[[directory_layout:gradle_user_home_directory]]
+== Gradle user home directory
+
+The Gradle user home directory has roughly the following structure:
+
+[listing]
+----
+.
+├── caches // <1>
+│   ├── 4.8 // <2>
+│   ├── 4.9 // <2>
+│   ├── ⋮
+│   ├── jars-3 // <3>
+│   └── modules-2 // <3>
+├── daemon // <4>
+│   ├── ⋮
+│   ├── 4.8
+│   └── 4.9
+├── init.d // <5>
+│   └── my-setup.gradle
+├── wrapper
+│   └── dists // <6>
+│       ├── ⋮
+│       ├── gradle-4.8-bin
+│       ├── gradle-4.9-all
+│       └── gradle-4.9-bin
+└── gradle.properties // <7>
+----
+<1> Global cache directory (for everything that's not project-specific)
+<2> Version-specific caches (e.g. to support incremental builds)
+<3> Shared caches (e.g. for artifacts of dependencies)
+<4> Registry and logs of the <<gradle_daemon.adoc#gradle_daemon, Gradle Daemon>>
+<5> Global <<init_scripts.adoc#init_scripts, initialization scripts>>
+<6> Distributions downloaded by the <<gradle_wrapper.adoc#gradle_wrapper,Gradle Wrapper>>
+<7> Global <<build_environment.adoc#sec:gradle_configuration_properties,Gradle configuration properties>>
+
+[[directory_layout:gradle_user_home_directory:cache_cleanup]]
+=== Global cache cleanup
```

Comment:
This transition is a little awkward. It feels like the document should be discussing some of the detail of the different directories. I would rather remove the "info" parts of the above annotations and insert a definition list after the directory structure that explains what each directory does and perhaps whether the user can garner any useful information from them.

I would also add a lead in paragraph for this section that says a little about what "Global cache cleanup" means. It's not a very descriptive heading on its own.

Reply:
An entry for the global cache directory could also link directly to the section on how it's cleaned up.